ui/StatusQ/CMakeLists.txt

```diff
@@ -2,29 +2,163 @@ cmake_minimum_required(VERSION 3.19)
 
 project(StatusQ)
 
-if (APPLE)
-    set(MACOS_VERSION_MIN_FLAGS -mmacosx-version-min=10.14)
-    set(CMAKE_OSX_ARCHITECTURES "x86_64")
-endif()
+option(STATUSQ_BUILD_SANDBOX "Enable to build StatusQ Sandbox application" ON)
+option(STATUSQ_BUILD_SANITY_CHECKER "Enable to build StatusQ Sanity Checker application" ON)
+option(STATUSQ_BUILD_TESTS "Enable to build StatusQ UI auto tests" ON)
 
 set(CMAKE_AUTOMOC ON)
 set(CMAKE_AUTORCC ON)
 
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 
 set(QZXING_USE_QML ON)
 set(QZXING_MULTIMEDIA ON)
 set(QZXING_USE_DECODER_QR_CODE ON)
 
-# https://doc.qt.io/qtcreator/creator-qml-modules-with-plugins.html#importing-qml-modules
-set(QML_IMPORT_PATH
-    ${CMAKE_SOURCE_DIR}/src;${QML_IMPORT_PATH}
-    CACHE STRING "")
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 
+find_package(Qt5 COMPONENTS
+        Core Qml Gui Quick QuickControls2 REQUIRED)
 
 add_subdirectory(../../vendor/SortFilterProxyModel SortFilterProxyModel)
```

Comment:
I know it's not directly related but referring to `SortFilterProxyModel` and `qzxing` from upper directories makes StatusQ dependent on external context (and sets us further from the goal of having StatusQ independent lib). Wouldn't it be better to keep it in the `vendor` directory of the `StatusQ` library?

Reply:
I also don't really like this way of adding subdirectories, but at the same time I agree with @jakubgs that it's good to have a single `vendor` directory: https://github.com/status-im/status-desktop/pull/9993

I think it doesn't really hurt here right now, but we're free to come to this discussion later, if it starts hurting 🙂 

wdyt?

Reply:
Yeah, it doesn't hurt now. For me if `StatusQ` is intended to be an independent library it could have own `vendor` for own dependencies. But I agree it's an additional complication. Alternatively we could remove dependency on `SortFilterProxyModel` at all. Anyways, it's ok for me to leave it as it is.

Reply:
I think scattering submodules across the whole repo will only lead to confusion, but I guess some people enjoy confusion.